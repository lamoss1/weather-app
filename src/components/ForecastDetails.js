// import React from "react";
// import "../styles/ForecastDetails.css";
// import "../styles/App.css";

// function ForecastDetails({ forecast }) {
//   const { date, humidity, temperature, wind } = forecast;
//   const formattedDate = new Date(date).toDateString();
//   return (
//     <div className="forecast-details">
//       <div className="forecast-details__date">{date}</div>
//       {formattedDate}
//       <div className="forecast-details_humidity">{humidity.max}</div>
//       <div className="forecast-details_temperature_min">{temperature.min}</div>
//       <div className="forecast-details_temperature_max">{temperature.max}</div>
//       <div className="forecast-details__wind">Wind: {wind.speed}mph</div>
//     </div>
//   );
// }

// export default ForecastDetails;

import React from "react";
import getPokemonImageUrl from "../helpers/getPokemonImageUrl";

function ForecastDetails(props) {
  const { date, temperature, description, windSpeed } = props;
  const pokemonImage = getPokemonImageUrl();
  // ...
  <img src={pokemonImage} alt="Pokemon" />;

  return (
    <div className="forecast-details">
      <h2>{new Date(date).toLocaleDateString()}</h2>
      <img src={pokemonImage} alt="Pokemon" />
      <p>
        <strong>Temperature:</strong> {temperature}Â°C
      </p>
      <p>
        <strong>Description:</strong> {description}
      </p>
      <p>
        <strong>Wind Speed:</strong> {windSpeed} mph
      </p>
    </div>
  );
}

export default ForecastDetails;
