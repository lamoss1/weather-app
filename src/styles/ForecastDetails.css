.forecast-details {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 2rem;
  color: #fff;
  font-size: 1.4rem;
  text-align: center;
}

.forecast-details__date {
  font-size: 3rem;
  margin-bottom: 1rem;
}

.forecast-details__temperature {
  font-size: 3rem;
  margin-bottom: 1rem;
}

.forecast-details__description {
  font-size: 2rem;
  margin-bottom: 1rem;
}

.forecast-details__icon {
  width: 10rem;
  margin-bottom: 1rem;
}

.forecast-details__humidity,
.forecast-details__wind-speed,
.forecast-details__wind-direction {
  font-size: 1.6rem;
  margin-top: 1rem;
}

.forecast-details__humidity span,
.forecast-details__wind-speed span,
.forecast-details__wind-direction span {
  font-weight: bold;
}

@media only screen and (min-width: 768px) {
  .forecast-details {
    font-size: 2rem;
  }

  .forecast-details__date {
    font-size: 4rem;
  }

  .forecast-details__temperature {
    font-size: 4rem;
  }

  .forecast-details__description {
    font-size: 3rem;
  }

  .forecast-details__icon {
    width: 15rem;
  }

  .forecast-details__humidity,
  .forecast-details__wind-speed,
  .forecast-details__wind-direction {
    font-size: 2rem;
  }
}
