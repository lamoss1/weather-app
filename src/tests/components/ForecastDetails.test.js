import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    humidity: {
      min: 0,
      max: 100,
    },
    wind: {
      speed: {
        min: 1,
        max: 100,
        direction: "stub",
      },
    },
    temperature: {
      min: 12,
      max: 22,
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails
        date={validProps.date}
        humidity={validProps.humidity}
        temperature={validProps.temperature}
        wind={validProps.wind}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
