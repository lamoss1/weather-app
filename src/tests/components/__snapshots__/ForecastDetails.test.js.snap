// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      1111111
    </div>
    Thu Jan 01 1970
    <div
      class="forecast-details_humidity"
    />
    <div
      class="forecast-details_temperature_min"
    >
      12
    </div>
    <div
      class="forecast-details_temperature_max"
    >
      22
    </div>
    <div
      class="forecast-details__wind"
    >
      Wind: 100mph
    </div>
  </div>
</DocumentFragment>
`;
